### ISSUE TITLE
Labels don't show unless the EOI is in it's place

### ISSUE THREAD

**Issue by mnbjhu** [zesterer/chumsky#631]
Suppose I have the following parser:

```rust
    let ident = select! {
        Token::Ident(ident) => ident,
    };
    just(Token::Keyword(Keyword::Select))
        .ignore_then(ident.labelled("column"))
        .then_ignore(just(Token::Keyword(Keyword::From)))
        .then(ident.labelled("table"))
        .map(|(column, from)| Statement::Select(Select { column, from }))
```
When I parse this input:

(Keyword(Select), 0..6)
(Ident("some"), 7..11)
(Keyword(From), 12..16)
(Newline, 16..17)

Then I get an error: 'found newline expected something else'. However if I use:

(Keyword(Select), 0..6)
(Ident("some"), 7..11)
(Keyword(From), 12..16)

Then I get the error: 'found end of input expected table' which is what I'd expect. Is this a bug or am I missing something? (1.0.0-alpha.7)

**Comment by zesterer** [comment]
Hmm, this odd and seems unintended. Do you have a minimal working example of this I can play with?

**Comment by mnbjhu** [comment]
Of course, I've created an example here: https://github.com/mnbjhu/label_test

**Comment by ojkelly** [comment]
I noticed this after attempting to update, and found it only occurs on sources without a newline. I'm working on a project with a few hundred tests without newlines (unit tests for individual parsers for example).

I had a dig through the diff from what I had been using and found what I think might be the cause https://github.com/zesterer/chumsky/pull/645

I'm not fully across the details of that area of code, so it may not be the ideal fix but hopefully it's useful.

**Comment by mnbjhu** [comment]
Thanks for this, I'll make sure to check it out this weekend. I've been working on a language server recently and this will be really useful :)
